appcues/src/main/java/com/appcues/RenderContextManager.kt

```diff
@@ -1,25 +1,80 @@
 package com.appcues
 
+import com.appcues.analytics.ExperienceLifecycleTracker
+import com.appcues.data.model.Experience
+import com.appcues.data.model.ExperienceTrigger
+import com.appcues.data.model.ExperienceTrigger.Qualification
 import com.appcues.data.model.RenderContext
 import com.appcues.data.model.RenderContext.Embed
+import com.appcues.statemachine.StateMachine
+import org.koin.core.component.KoinScopeComponent
+import org.koin.core.component.get
+import org.koin.core.component.inject
+import org.koin.core.scope.Scope
 import java.lang.ref.WeakReference
 
-internal class RenderContextManager {
+internal class RenderContextManager(override val scope: Scope) : KoinScopeComponent {
 
-    private val slots: HashMap<RenderContext, WeakReference<AppcuesFrameView>> = hashMapOf()
+    private val lifecycleTracker by inject<ExperienceLifecycleTracker>()
+
+    private val frameSlots: HashMap<RenderContext, WeakReference<AppcuesFrameView>> = hashMapOf()
+
+    private val stateMachineSlots: HashMap<RenderContext, StateMachine> = hashMapOf()
```

Comment:
just noting, on the iOS side, the `AppcuesFrameView` actually holds the reference to the `StateMachine`, keeping that one-to-one relationship defined there rather than having to have the two different maps here, ref: https://github.com/appcues/appcues-ios-sdk/blob/b074f690978ccb622bc7c88da8499b613a8002ac/Sources/AppcuesKit/Presentation/UI/AppcuesFrameView.swift#L19-L24

I don't know if that simplifies things here at all, or would cause issue, but wanted to at least note it if you thought it might make things easier to manage.

Reply:
I may need to have the reference there for what I'm trying to do, so good to know!